File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    rows = len(arr) #행의수
    cols = len(arr[0]) #열의수

    # 행의 수와 열의 수가 같다면 arr을 그대로 반환
    if rows == cols:
        return arr
    
    # 행의 수가 더 많은 경우
    if rows > cols:
        for i in range(rows):
            arr[i] += [0] * (rows - cols)
        return arr

    # 열의 수가 더 많은 경우
    else:
        # 각 열의 끝에 0을 추가
        for i in range(cols -rows):
            arr.append([0] * cols)
        return arr

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int rows = arr.length;
        int cols = arr[0].length;
        int size = Math.max(rows,cols);
        // 행이 열보다많으면 열이 적은만큼 0을 채우고
        int[][] answer = new int[size][size];
        
        // 열이 행보다 많으면 행이 적은 만큼 0으로 채운다.
        for(int i=0;i<rows;i++){
            for(int j=0; j<cols;j++){
                answer[i][j]=arr[i][j];
            }
        }
        
        return answer;
    }
}